### PR TITLE
Mypy and test fixes, black formatting

### PR DESCRIPTION
**Pull request by Rigidity**
This fixes mypy errors, type stubs, and formats all of the test files with black. Black formatting is checked in CI as well.

You can review one commit at a time, and mostly ignore the "Black formatting" commit, since no changes were made in it other than formatting and adding CI to enforce the formatting.

### CODE REVIEW

File: wheel/chia_rs.pyi [hunk 1]
```diff
@@ -55,19 +55,19 @@ NO_UNKNOWN_OPS: int = ...
...
 def run_chia_program(
     program: bytes, args: bytes, max_cost: int, flags: int
-) -> Pair[int, LazyNode]: ...
+) -> Tuple[int, LazyNode]: ...
 
 class LazyNode:
-    def pair() -> Optional[Tuple[LazyNode, LazyNode]]: ...
-    def atom() -> bytes: ...
+    pair: Optional[Tuple[LazyNode, LazyNode]]
+    atom: bytes
```
Comment (arvidn) [inline wheel/chia_rs.pyi:62]: actually, it looks like this should be `Optional[bytes]`. Did some tool detect this and say it should be just `bytes`?

File: wheel/chia_rs.pyi [hunk 2]
```diff
@@ -91,11 +91,8 @@ class G1Element:
     def get_fingerprint(self) -> int: ...
     def pair(self, other: G2Element) -> GTElement: ...
     @staticmethod
-    def from_bytes_unchecked(b: bytes) -> G1Element: ...
-    @staticmethod
     def generator() -> G1Element: ...
     def __str__(self) -> str: ...
-    def __repr__(self) -> str: ...
```
Comment (arvidn) [inline wheel/chia_rs.pyi:98]: are all the `__repr__()` and `from_bytes_unchecked()` that you're removing duplicates?

**Review by arvidn — COMMENTED** [review]
it all looks good, except the type of the `atom` property should be `Optional[bytes]` as far as I can tell

File: wheel/chia_rs.pyi (hunk 2 above)
Comment (Rigidity) [reply wheel/chia_rs.pyi:98]: yes

File: wheel/chia_rs.pyi (hunk 1 above)
Comment (Rigidity) [reply wheel/chia_rs.pyi:62]: ok I'll fix that in a bit, the generate type stubs script hard codes this (another reason automatic generation would be nice for everything later)

**Review by arvidn — APPROVED** [review]

**Pull request merged**